### PR TITLE
Add timer metric for total database duration

### PR DESCRIPTION
**Pull request by siladu**
Fixes https://github.com/ConsenSys/web3signer/issues/561

### CODE REVIEW

File: core/src/main/java/tech/pegasys/web3signer/core/metrics/SlashingProtectionMetrics.java [hunk 1]
```diff
@@ -34,6 +36,12 @@ public SlashingProtectionMetrics(final MetricsSystem metricsSystem) {
             ETH2_SLASHING_PROTECTION,
             "prevented_signings",
             "The number of prevented signings due to violation of slashing conditions.");
+
+    this.databaseTimer =
+        metricsSystem.createTimer(
+            ETH2_SLASHING_PROTECTION,
+            "slashing_protection_database_duration",
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/metrics/SlashingProtectionMetrics.java:43]: slashing_protection prefix can be removed. The metrics name is appended with the metric category so will already have a "eth2_slashingprotection" prefix

File: core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java [hunk 2]
```diff
@@ -149,7 +149,7 @@ private void signWithSlashingProtection(
       final Eth2SigningRequestBody eth2SigningRequestBody,
       final Bytes signingRoot,
       final String signature) {
-    try {
+    try (final TimingContext ignored = slashingMetrics.getDatabaseTimer().startTimer()) {
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java:152]: Would be better to put this in the DbSlashingProtection class so it's as close as possible to the time taken for the db queries.

File: core/src/main/java/tech/pegasys/web3signer/core/metrics/SlashingProtectionMetrics.java [hunk 3]
```diff
@@ -34,6 +36,12 @@ public SlashingProtectionMetrics(final MetricsSystem metricsSystem) {
             ETH2_SLASHING_PROTECTION,
             "prevented_signings",
             "The number of prevented signings due to violation of slashing conditions.");
+
+    this.databaseTimer =
+        metricsSystem.createTimer(
```
Comment (jframe) [inline core/src/main/java/tech/pegasys/web3signer/core/metrics/SlashingProtectionMetrics.java:41]: Would be good to create a timer with a label so we can database queries broken down into attestation and block duration times.

**Comment by siladu** [comment]
Put the metrics as close to the DB as I could without crossing into the slashing module, which would require a larger refactor with the only practical benefit being excluding one non-DB related code...although I appreciate that could change in the future so maybe the inside the slashing module is still the right place.

**Review by jframe — APPROVED** [review]
LGTM

**Pull request merged**